src/proj.rs

```diff
@@ -1286,6 +1288,142 @@ impl Proj {
             }
         }
     }
+
+    pub fn as_wkt(
+        &self,
+        version: Option<WktVersion>,
+        options: Option<WktOptions>,
+    ) -> Result<String, ProjError> {
+        let options_str = if let Some(ref options) = options {
+            let mut opts = vec![];
+            if let Some(multiline) = options.multiline {
+                opts.push(CString::new(format!(
+                    "MULTILINE={}",
+                    if multiline { "YES" } else { "NO" }
+                ))?)
+            };
+
+            if let Some(indentation_width) = options.indentation_width {
+                opts.push(CString::new(format!(
+                    "INDENTATION_WIDTH={}",
+                    indentation_width
+                ))?)
+            }
+
+            if let Some(ref output_axis) = options.output_axis {
+                opts.push(CString::new(format!(
+                    "OUTPUT_AXIS={}",
+                    match output_axis {
+                        WktOutputAxis::Auto => "AUTO",
+                        WktOutputAxis::Yes => "YES",
+                        WktOutputAxis::No => "NO",
+                    }
+                ))?);
+            }
+
+            if let Some(strict) = options.strict {
+                opts.push(CString::new(format!(
+                    "STRICT={}",
+                    if strict { "YES" } else { "NO" }
+                ))?);
+            }
+
+            if let Some(allow_ellipsoidal_height_as_vertical_crs) =
+                options.allow_ellipsoidal_height_as_vertical_crs
+            {
+                opts.push(CString::new(format!(
+                    "ALLOW_ELLIPSOIDAL_HEIGHT_AS_VERTICAL_CRS={}",
+                    if allow_ellipsoidal_height_as_vertical_crs {
+                        "YES"
+                    } else {
+                        "NO"
+                    }
+                ))?);
+            }
+
+            if let Some(allow_linunit_node) = options.allow_linunit_node {
+                opts.push(CString::new(format!(
+                    "ALLOW_LINUNIT_NODE={}",
+                    if allow_linunit_node { "YES" } else { "NO" }
+                ))?);
+            }
+
+            if opts.is_empty() {
+                None
+            } else {
+                Some(opts)
+            }
+        } else {
```

Comment:
Only a style nit, so take it or leave it.

Rather than having this far away dangling else, you could:
```
let options_str = options.as_ref().map(|options| {
   ...
});
```

Reply:
There are question mark operators on the CString::new() method that won't return if I wrap the code in a closure. Atleast not without adding a transpose() on the end. I'd be happy to implement that if you prefer.